src/foam/mlang/mlangJava.js

```diff
@@ -323,6 +323,20 @@ foam.CLASS({
 });
 
 
+foam.CLASS({
+  refines: 'foam.mlang.predicate.Has',
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: 'Object value = getArg1().f(obj); '+
+        'return ! (value == null || '+
```

Comment:
We can use getClass().isArray() && java.lang.reflect.Array.getLength(value)

**https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/Array.html#getLength(java.lang.Object)**